### PR TITLE
BUG: Parse missing values using read_json with dtype=False to NaN instead of None (GH28501)

### LINKED ISSUE #28501: read_json with dtype=False infers Missing Values as None
Run against master:

```python
In [13]: pd.read_json("[null]", dtype=True)
Out[13]:
    0
0 NaN

In [14]: pd.read_json("[null]", dtype=False)
Out[14]:
      0
0  None
```

I think the second above is an issue - should probably return `np.nan` instead of `None`

### PR DESCRIPTION
**Pull request by avinashpancham**
- [x] closes #28501 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — APPROVED** [review]
yep this is way more consistent, thanks @avinashpancham 
cc @WillAyd merge when ready

**Comment by WillAyd** [comment]
Nice PR! Looks like there is a merge conflict in the whatsnew - if you can fix that up we can get this merged

**Comment by avinashpancham** [comment]
Merged master, but CI fails due to unrelated reasons. Will retry with new updates from master later today

**Comment by jreback** [comment]
@avinashpancham can you merge master once again

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -624,6 +624,7 @@ I/O
 - Bug in :func:`read_html` was raising a ``TypeError`` when supplying a ``pathlib.Path`` argument to the ``io`` parameter (:issue:`37705`)
 - :meth:`to_excel` and :meth:`to_markdown` support writing to fsspec URLs such as S3 and Google Cloud Storage (:issue:`33987`)
 - Bug in :meth:`read_fw` was not skipping blank lines (even with ``skip_blank_lines=True``) (:issue:`37758`)
+- Parse missing values using :func:`read_json` with ``dtype=False`` to NaN instead of None (:issue:`28501`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:627]: use double backticks around ``NaN`` and ``None``

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (avinashpancham) [reply doc/source/whatsnew/v1.2.0.rst]: Done

**Comment by avinashpancham** [comment]
@jreback merged master and updated whatsnew entry

**Comment by jreback** [comment]
hmm precommit checks are failing, can you merge master and ping on green

**Comment by avinashpancham** [comment]
Still some issues, will try again tomorrow

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @avinashpancham

**Comment by simonjayhawkins** [comment]
I'm not sure that there was consensus in the issue discussion on the correct behaviour or whether it is a bug. Maybe worth considering moving the release note into the breaking changes section in case anyone is depending on this behaviour.

**Comment by jreback** [comment]
> I'm not sure that there was consensus in the issue discussion on the correct behaviour or whether it is a bug. Maybe worth considering moving the release note into the breaking changes section in case anyone is depending on this behaviour.

type dtype option is actually pretty odd anyhow (and should deprecate)

**Comment by jorisvandenbossche** [comment]
I think most voices in the issue actually argued that the current behaviour was correct, and so IMO we should not have changed it to start with.

**Comment by jreback** [comment]
there was exactly 1 comment

**Comment by jorisvandenbossche** [comment]
Will opened the issue, and 2 people commented on it arguing not to change it (and a third one commented as well, but off-topic), no one else argued in favor of the change.

Anyway, I don't think counting the numbers is that important, but I agree with @simonjayhawkins that there was certainly no clear consensus

**Comment by jorisvandenbossche** [comment]
On the None vs NaN, I am personally quite ambivalent, but when disabling parsing I think we should *not* return float dtype for this case.

**Comment by jreback** [comment]
you can revert if u want
not that important 

this was removing a special case though - it was very odd

**Pull request merged**